### ISSUE TITLE
Should ~/.profile be sourced (indirectly) from ~/.zshrc?

### ISSUE THREAD

**Issue by jw-00000** [ohmyzsh/ohmyzsh#3411]
I'm not sure this is a bug, so bear with me for a minute here.

The default template for ~/.zshrc includes oh-my-zsh.sh, which in turn calls tools/check_for_upgrade.sh, which sources ~/.profile. This means that for every terminal that gets created, ~/.profile is called.

However, in my understanding (see e.g. [1](http://www.tldp.org/LDP/Bash-Beginners-Guide/html/sect_03_01.html), [2](http://serverfault.com/a/261807/112267), [3](http://mywiki.wooledge.org/DotFiles)), ~/.profile should only be executed once, on login, while ~/.zshrc is called each time a terminal window is created.

In my case concretely, this causes a problem as my ~/.profile calls xmodmap (I'm trying to configure my caps lock key as an extra backspace as per [1](http://askubuntu.com/a/82734/4335)), which doesn't like being called multiple times. In general, this might cause problems if ~/.profile contains long-running processes or code that assumes it will only be called on login.

So, to summarize: should ~/.profile be sourced in tools/check_for_upgrade.sh, and therefore be called for each terminal window that is created?

Additional info: I also tried to find out where the line sourcing ~./profile got introduced, it was added in [this commit](https://github.com/robbyrussell/oh-my-zsh/commit/a9e1d9aa890c18ea3c3250442e7d9de555633237). I dug a bit deeper to find out why it was added, which lead to [this commit](https://github.com/n-st/oh-my-zsh/commit/3f44f51e9ca1b196e4c3caa439558996abdac273), with commit message "source ~/.profile for upgrading (to source the proxy configuration)".
